### ISSUE TITLE
Speed up searching for exclusions

### ISSUE THREAD

**Issue by jmarceli** [stevegrunwell/asimov#19]
Thanks for that tool, exactly something I was looking for.

One thing that may make it even better would be skipping some common folders while searching for development dependencies like:

- `~/Library`
- `~/Applications`
- `~/Music`
etc.

**Comment by rowanbeentje** [comment]
https://github.com/stevegrunwell/asimov/pull/18 excludes `~/Library` as part of other speedup work - we can add others in the same way I suppose, but I wonder if there's a point where the exclusion checks make the overall process slower than just quickly scanning directories!

**Comment by jmarceli** [comment]
I was thinking rather about exclusion directly through `find`, something like (according to https://stackoverflow.com/questions/4210042/how-to-exclude-a-directory-in-find-command):

```bash
find ~ -type d -name "${parts[0]}" -not -path "/Users/*/Library/*"
```

I think that this shouldn't hit overall performance (I'm not a `find` expert so I may be wrong).

Also `~/.Trash` exclusion would be nice, because even if I have trash excluded in TimeMachine I still have to wait for Asimov to scan through.

Regarding the speed. 
Maybe excluding nested `node_modules` with `find` rather than inside the `dependency_file_exists()`  would save some time?
What I mean is:
```bash
 find ~ -type d -name "${parts[0]}" -not -path "*/node_modules/*/node_modules*"
```

**Comment by rowanbeentje** [comment]
https://github.com/stevegrunwell/asimov/pull/17 (included in https://github.com/stevegrunwell/asimov/pull/18 as well!) excludes subdeps, and yeah, that `-not -path` is the approach used to exclude `~/Library` - I just can't help feeling that will have a cost.

(Trashes folders are a great one though)

**Comment by jmarceli** [comment]
I'm still waiting for `~/.Trash` to complete processing (about 20min) despite the fact that I've excluded trash in TM anyway...

The first run of Asimov takes about 2:30 hours, but the overall effect is great.
Thanks for this script.

**Comment by FlorianWendelborn** [comment]
> Also ~/.Trash exclusion would be nice, because even if I have trash excluded in TimeMachine I still have to wait for Asimov to scan through.

Wouldn't it be better to just automatically skip all folders that are excluded from time machine then?

**Comment by muuvmuuv** [comment]
Yep, why not exclude everything inside the TimeMaschine ignore list from the find process? I mean the ignore list that you can manually edit in the TimeMaschin pane in settings.

**Comment by django23** [comment]
This got a *lot* of love @jmarceli 🚀

`~/Library` and `~/.Trash` have been pruned for a while, and v0.6.0 added a persistent path cache plus incremental Spotlight (`mdfind`) discovery — so repeat runs skip the filesystem traversal entirely. Typical home directory: **~75s → ~1–2s**.

`--full-scan` forces a fresh traversal when you want one. Closing as very much fixed. 🙏
